File: 035_MaxConsecutiveOnes.py
"""
Problem: Max Consecutive Ones
Description: Find the maximum number of consecutive 1s in a binary array.

Approach: Use a counter to track consecutive ones, update max when 0 is encountered.
"""

from typing import List

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_count = 0
        current_count = 0
        
        for num in nums:
            if num == 1:
                current_count += 1
                max_count = max(max_count, current_count)
            else:
                current_count = 0
        
        return max_count

if __name__ == "__main__":
    sol = Solution()
    nums = [1, 1, 0, 1, 1, 1]
    print(f"Max Consecutive Ones: {sol.findMaxConsecutiveOnes(nums)}")

"""
Time Complexity: O(n) where n is the length of the array
Space Complexity: O(1)
"""

File: 035_MaxConsecutiveOnes.cpp
/*
Problem: Max Consecutive Ones
Description: Find the maximum number of consecutive 1s in a binary array.

Approach: Use a counter to track consecutive ones, update max when 0 is encountered.
*/

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxCount = 0;
        int currentCount = 0;
        
        for (int num : nums) {
            if (num == 1) {
                currentCount++;
                maxCount = max(maxCount, currentCount);
            } else {
                currentCount = 0;
            }
        }
        
        return maxCount;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {1, 1, 0, 1, 1, 1};
    cout << "Max Consecutive Ones: " << sol.findMaxConsecutiveOnes(nums) << endl;
    
    return 0;
}

/*
Time Complexity: O(n) where n is the length of the array
Space Complexity: O(1)
*/
